### PR TITLE
Add support for creating/managing a ComputeBuffer through scripting

### PR DESCRIPTION
**Pull request by jsundra**
Created a new `GraphicsJsInterface` preview scripting API, as well as a shared `ScriptableGraphics` class for graphics/rendering related scripting APIs to use.

Graphics currently allows for creating/destroying `ComputeBuffer`s, with the `MaterialJsApi` getting a small tweak to allow setting a buffer by id.

Example behavior script:
```
const graphics = require('graphics-preview');
if (!graphics) return;

var bufferId;

function enter() {
  if (!this.material) {
    log.error('No material found for the Canvas to use.');
    return;
  }
  
  bufferId = graphics.createBuffer(1024, 4 * 3); // 1024x1024, float3
  graphics.setWriteTarget(bufferId, 1);
}

function exit() {
  if (bufferId) {
    graphics.clearWriteTargets();
    graphics.destroyBuffer(bufferId);
  }
}

module.exports = {
  enter: enter,
  exit: exit
};
```

Also, started adding new features in the Enklu namespace, figured why not?!

### CODE REVIEW

**Comment by gdamron** [comment]
Did you forget to commit the actual JS interface?

**Comment by jsundra** [comment]
Well that's just embarrassing.

**Comment by jsundra** [comment]
I'll add comments too ;)

File: Runtime/Source/Player/Scripting/Interface/Elements/MaterialJsApi.cs [hunk 1]
```diff
@@ -246,6 +251,16 @@ public void setColor(string param, Col4 value)
             }
         }
 
+        public void setBuffer(string param, int bufferId)
```
Comment (gdamron) [inline Runtime/Source/Player/Scripting/Interface/Elements/MaterialJsApi.cs:254]: comments!

File: Runtime/Source/Player/Scripting/Interface/Graphics/GraphicsJsInterface.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            _graphics = graphics;
+        }
+        
+        /// <summary>
+        /// Creates a ComputeBuffer.
+        /// </summary>
+        /// <param name="size">The size of a square texture, in one dimension (1024 for 1024x1024).</param>
+        /// <param name="stride">The stride of the buffer (Typically 12 or 16, 4byte float * RGB / RGBA)</param>
+        /// <returns>The id of the created buffer.</returns>
+        public int createBuffer(int size, int stride)
```
Comment (gdamron) [inline Runtime/Source/Player/Scripting/Interface/Graphics/GraphicsJsInterface.cs:33]: wondering if it's worth logging a warning or something if size isn't a power of 2

Comment (jsundra) [reply Runtime/Source/Player/Scripting/Interface/Graphics/GraphicsJsInterface.cs:33]: Ah good idea, I should also check against 0 & negative numbers too!

**Pull request merged**